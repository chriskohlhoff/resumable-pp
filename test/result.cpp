#include <stdio.h>
#include <cassert>

class heavy
{
public:
  heavy(const char* s) { printf("heavy(%s)\n", s); }
  heavy(const heavy&) { printf("heavy(const heavy&)\n"); }
  ~heavy() { printf("~heavy()\n"); }
  void print() { printf("here we are\n"); }
};

int main()
{
  auto&& h = [=] resumable {
    return heavy("foo");
  };

  printf("before call\n");
  resume(h);
  printf("after call\n");
  assert(ready(h));
  result(h).print();
}
