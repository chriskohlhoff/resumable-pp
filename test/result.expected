before call
heavy(foo)
after call
heavy(const heavy&)
here we are
~heavy()
~heavy()
